tls.c

```diff
@@ -162,6 +162,17 @@ int ssl_init(void) {
         exit(EX_USAGE);
     }
 
+    // Optional session caching; default disabled.
+    if (settings.ssl_session_cache) {
+        SSL_CTX_sess_set_new_cb(settings.ssl_ctx, ssl_new_session_callback);
```

Comment:
Is there a reason why we don't need/can't have the `ssl_new_sessions` counter when the session cache is not enabled? I guess this will be same as the number of connections made in that case.

Reply:
The set-session callback isn't actually registered internally in OpenSSL unless session caching is enabled. So when the counter is enabled when session caching is off, it stays at 0. This behavior seems consistent with the documentation: https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_sess_set_new_cb.html